File: Algorithms/src/1513. Number of Substrings With Only 1s/Code/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int numSub(string s)
    {
        const long long MOD = 1000000007LL;
        long long res = 0;
        long long cnt = 0; // current consecutive '1's count

        for (char c : s)
        {
            if (c == '1')
            {
                cnt++;
            }
            else
            {
                // block ended, add contribution
                res = (res + (cnt * (cnt + 1) / 2) % MOD) % MOD;
                cnt = 0;
            }
        }
        // add last block if string ended with '1'
        res = (res + (cnt * (cnt + 1) / 2) % MOD) % MOD;
        return (int)res;
    }
};

File: Algorithms/src/1513. Number of Substrings With Only 1s/Code/solution.py
class Solution:
    def numSub(self, s: str) -> int:
        MOD = 10**9 + 7
        res = 0
        cnt = 0  # current consecutive '1's count

        for ch in s:
            if ch == '1':
                cnt += 1
            else:
                res = (res + (cnt * (cnt + 1) // 2) % MOD) % MOD
                cnt = 0

        # add last block if any
        res = (res + (cnt * (cnt + 1) // 2) % MOD) % MOD
        return res
